nilearn/plotting/displays/_figures.py

```diff
@@ -104,3 +130,355 @@ def savefig(self, output_file=None):
+                x = np.average((t0[0], t1[0], t2[0]), weights=(w0, w1, w2))
+                y = np.average((t0[1], t1[1], t2[1]), weights=(w0, w1, w2))
+                z = np.average((t0[2], t1[2], t2[2]), weights=(w0, w1, w2))
+                centroids.append(
+                    self._project_above_face(
+                        np.array((x, y, z)), t0, t1, t2, elevation=elevation
+                    )
+                )
+                segs = [None] * 3
+                if face[0] in parc_idx and face[1] in parc_idx:
+                    segs[0] = self._transform_coord_to_plane(
+                        t0, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t1, t0, t1, t2)
+                if face[0] in parc_idx and face[2] in parc_idx:
+                    segs[1] = self._transform_coord_to_plane(
+                        t0, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t2, t0, t1, t2)
+                if face[1] in parc_idx and face[2] in parc_idx:
+                    segs[2] = self._transform_coord_to_plane(
+                        t2, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t1, t0, t1, t2)
+                segments.append(tuple(segs))
+                vs.append((t0, t1, t2))
+                idxs.append([f for f in face if f in parc_idx])
+
+        centroids = np.array(centroids)
+
+        # Next, sort centroids along boundary
+        # Start with the first vertex
+        current_vertex = 0
+        visited_vertices = {current_vertex}
+        last_distance = np.inf
+        prev_first = 0
+
+        sorted_vertices = [centroids[0]]
+
+        # Loop over the remaining vertices in order of distance from the
+        # current vertex
+        while len(visited_vertices) < len(centroids):
+            remaining_vertices = np.array(
+                [
+                    vertex
+                    for vertex in range(len(centroids))
+                    if vertex not in visited_vertices
+                ]
+            )
+            remaining_distances = distance_matrix(
+                centroids[current_vertex].reshape(1, -1),
+                centroids[remaining_vertices],
+            )
+            # Occasionally, the next closest centroid is one that would
+            # cause a loop. This is common when a vertex is a neighbor
+            # of only one other vertex in the roi (the loop encircles
+            # this corner vertex). So, the next added centroid is one
+            # that may be slightly farther away -- if the one that is
+            # farther away has fewer vertices within the roi.
+            if len(remaining_vertices) > 5:
+                smallest_5_idx = np.argpartition(
+                    remaining_distances.squeeze(), 5
+                )[:5]
+                xy1 = self._transform_coord_to_plane(
+                    centroids[current_vertex], *vs[current_vertex]
+                )
+                next_index = 9999
```

Comment:
is this initialization safe ?

Reply:
Oh, I suppose it would be a problem if there were 9999 regions within a parcellation--changing this to -1

Reply:
[a50d656](https://github.com/nilearn/nilearn/pull/3949/commits/a50d65639b9195f40bee9ee6d2abb46e7536c22b)